### DISCUSSION TITLE
In Segmentation tasks, whether have a strategy to keep pixel int type after kornia.geometry.transform.warp_perspective?

### DISCUSSION THREAD

**Discussion by raphaelhou25** [kornia/kornia#2434] [Questions]
my source image:

![image](https://github.com/kornia/kornia/assets/37241024/6eaa820d-92a0-4077-aab3-68d646936dec)

in this image, all pixels in every line is marked by a int number, eg. 1,2,3,4,5.

homograph matrix is :

[[ 1.9455e-02,  4.6595e-01,  2.0739e+01],
  [ 3.5880e-01,  1.6237e-01, -6.6169e+00],
  [ 7.4005e-03,  2.4014e-03, -1.1517e-01]]

after kornia.geometry.transform.warp_perspective, all pixel values changed to float.

![image](https://github.com/kornia/kornia/assets/37241024/e8897a9b-8f5a-4a0e-80a4-30b938d82feb)

whether have a strateger to keep pixels int type after kornia.geometry.transform.warp_perspective?

i am looking forward to your reply, thx.

**Comment by edgarriba** [comment]
hi! maybe you could try the following:

1. convert to floating point
2. warp with nearest interpolation
3. convert the result back to int
